src/strand.rs

```diff
@@ -0,0 +1,306 @@
+}
+
+impl Same for ReqStrand {
+    fn same(&self, s1: &Self) -> bool {
+        self == s1
+    }
+}
+
+impl Display for ReqStrand {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        write!(f, "({})", self.strand_symbol())
+    }
+}
+
+impl Strandedness for ReqStrand {
+    fn reverse(&self) -> Self {
+        match *self {
+            ReqStrand::Forward => ReqStrand::Reverse,
+            ReqStrand::Reverse => ReqStrand::Forward,
+        }
+    }
+
+    fn try_req_strand(&self) -> Option<ReqStrand> {
+        Some(*self)
+    }
+
+    fn strand_symbol(&self) -> &str {
+        match *self {
+            ReqStrand::Forward => "+",
+            ReqStrand::Reverse => "-",
+        }
+    }
+
+    fn break_pos_strand(posstr: &str) -> Option<(&str, Self)> {
+        if posstr.ends_with("(+)") {
+            if let Some(pos) = posstr.get(0..(posstr.len() - 3)) {
+                return Some((pos, ReqStrand::Forward));
+            }
+        } else if posstr.ends_with("(-)") {
+            if let Some(pos) = posstr.get(0..(posstr.len() - 3)) {
+                return Some((pos, ReqStrand::Reverse));
+            }
+        }
+
+        None
+    }
+}
+
+impl KnownStrandedness for ReqStrand {
+    fn req_strand(&self) -> ReqStrand {
+        *self
+    }
+}
+
+/// Strand information for annotations that definitively have no
+/// strand information.
+#[derive(Debug, Clone, Hash, PartialEq, Eq, Ord, PartialOrd, Copy)]
+pub enum NoStrand {
+    Unknown,
+}
+
+impl Same for NoStrand {
+    fn same(&self, _s1: &Self) -> bool {
+        true
+    }
+}
+
+impl Display for NoStrand {
+    fn fmt(&self, _f: &mut Formatter) -> fmt::Result {
+        Ok(())
+    }
+}
+
+impl Strandedness for NoStrand {
+    fn reverse(&self) -> Self {
+        NoStrand::Unknown
+    }
+    fn try_req_strand(&self) -> Option<ReqStrand> {
+        None
+    }
+    fn strand_symbol(&self) -> &str {
+        "."
+    }
+    fn break_pos_strand(posstr: &str) -> Option<(&str, Self)> {
+        Some((posstr, NoStrand::Unknown))
+    }
+}
+
+/// Equality-like operator for comparing strand information. Unknown
+/// strands are not equal, but they are the "same" as other unknown
+/// strands.
+pub trait Same {
+    /// Indicate when two strands are the "same" -- two
+    /// unknown/unspecified strands are the "same" but are not equal.
+    fn same(&self, &Self) -> bool;
+}
+
+impl<T> Same for Option<T>
+where
+    T: Same,
+{
+    fn same(&self, s1: &Self) -> bool {
+        match (self, s1) {
+            (&Option::None, &Option::None) => true,
+            (&Option::Some(ref x), &Option::Some(ref x1)) => x.same(x1),
+            (_, _) => false,
+        }
+    }
+}
+
+/// Strand information for annotations on sequences
+pub trait Strandedness {
+    /// Reversed strand
+    fn reverse(&self) -> Self;
+
+    /// Convert into a `ReqStrand`, indicating a specific, known
+    /// strand. The return is wrapped in an `Option`, so unknown
+    /// strands will return `None`.
```

Comment:
Please add an example here. From the description and the name it is not clear to me what this is supposed to do.

Reply:
Now I get it. Still, docs could be expanded. I wonder if it makes rather sense to implement `Into<Option<ReqStrand>>` for `Strand`? This could enable some automatic conversion functions from rust and seems to be more canonical.

Reply:
As you suggest, I changed this to handle it entirely with From<...> traits.